base-template/contributors-guide.md

```diff
@@ -1,51 +1,250 @@
+You need to create a fork of the repository you want to contribute to.
+A fork copies the current state of the repository to your own project.
+When you have made your changes, you can propose the changes back to the original project with a pull request.
+The community reviews your pull request, and you can make more updates or changes as needed.
+When everyone agrees, the pull request gets merged, and your changes become part of the code.
+
+This example uses the `incubator` repository, but the method is the same for any repository.
+If you need more help with git, check out the github documentation at [https://docs.github.com/].
+For a glossary of git terms, see [The git glossary](https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitglossary.html)
+
+#### Creating a fork
+
+1. Make sure you have a [github](github.com) account, and that you're signed in.
+1. Navigate to the [Good Docs Project](https://github.com/thegooddocsproject), and click the repository you want to fork.
+1. Click the `Fork` button in the top-right corner, and select the account you want to use.
+1. Wait for github to create your fork and redirect you.
+Take note of the repository you are now in, called `github.com/<your_username>/incubator`.
+
+When you have created the fork, you can clone the code to your local machine.
+
+#### Cloning the code
+
+1. Create a directory on your local filesystem to keep your work in.
+You can call it anything you like, but most people use `workspace` or something similar.
+1. In your browser, go to the github page for your fork.
+The URL will be `github.com/<your_username>/incubator`, or you can get to it by clicking on your user image from the main github page, and selecting `Your repositories`.
+1. Click the green `Code` download button, and copy the URL in the drop-down box.
+1. Open a terminal on your local machine, make sure you are in your `workspace` directory, and clone the URL you copied.
+Depending on how big the repository is, it could take some time to download.
+
+   ```bash
+   > cd ~/workspace
+   > git clone https://github.com/<your_username>/incubator.git
+   ```
+
+When you have the code downloaded, you need to set the upstream repository.
+This is important so that you can keep your local fork updated.
+
+#### Setting the upstream repository
+
+1. In your terminal, navigate to the directory where you checked out the code:
+
+   ```bash
+   > cd ~/workspace/incubator
+   ```
+
+1. Check out the `master` branch:
+
+   ```bash
+   > git checkout master
+   ```
+
+1. List the current remote branches:
+
+   ```bash
+   > git remote -v
+   ```
+
+   This command should list two remotes, both marked `origin`, like this:
```

Comment:
point out that `origin` is __YOU!! 😁 __ that is your fork you can do whatever you want there.